tap_salesforce/__init__.py

```diff
@@ -116,6 +117,18 @@ def do_discover(sf):
 
     # For each SF Object describe it, loop its fields and build a schema
     entries = []
+
+    # Check if the user has BULK API enabled
+    if sf.api_type == 'BULK':
+        try:
+            bulk = Bulk(sf)
+            bulk.check_bulk_quota_usage()
+        except requests.exceptions.HTTPError as err:
+            if err.response is not None and len(err.response.json()) > 0:
+                for error_response_item in err.response.json():
+                    if error_response_item.get('errorCode') == 'API_DISABLED_FOR_ORG':
+                        raise TapSalesforceBulkAPIDisabledException('This client does not have Bulk API permissions, received "API_DISABLED_FOR_ORG" error code')
```

Comment:
Perhaps this block belongs in the bulk module and could be "what happens" when you create an instance.